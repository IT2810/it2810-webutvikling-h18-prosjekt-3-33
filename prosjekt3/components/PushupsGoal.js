import React from 'react';
import {
  View, Text, StyleSheet, ScrollView, Button, TouchableOpacity

} from 'react-native';
import Colors from '../constants/Colors.js';
import PushupsCounter from './PushupsCounter';

export default class PushupsGoal extends React.Component {
  state = {
    pushupsGoal: 0,
  }
  componentDidMount(){
    this.setState({pushupsGoal: this.props.pushupsGoal})
  }
  componentDidUpdate(prevProps, prevState){
    if(this.props.studyGoal != prevProps.studyGoal){
      this.setState({pushupsGoal: this.props.pushupsGoal})
    }
  }
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of pushups today </Text>
          <PushupsCounter pushupsGoal={this.props.pushupsGoal} />
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.goalComponentOne,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
