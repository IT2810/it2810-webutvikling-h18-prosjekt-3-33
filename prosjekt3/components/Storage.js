import React from 'react';
import {
    AsyncStorage,
} from 'react-native';


/* Format of the storage:


{ CONTACTS:   [ { name: "Ulrik", age: 22, address: "Bakklandet"  },
                { name: "Sondre", age: 24, address: "Ila"        },
                { name: "Jørgen", age: 95, address: "Møhlenberg" }, 
                ....
              ],
  

  TASKS:      [ { title: "Meeting", date: "22-12-1995" },
                { title: "Dinner", date: "12-7-2995" },

                ...

              ]

  GOALS:      [ {title: "Stop smoking", duration: 5, start: "22-12-1995"}

              ] 
}





*/