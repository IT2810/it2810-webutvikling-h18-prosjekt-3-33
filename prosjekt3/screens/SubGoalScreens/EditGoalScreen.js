import React from 'react';
import {
	View,
	Text,
	TouchableOpacity,
	Button,
	StyleSheet
} from 'react-native';


export default class EditGoalScreen extends React.Component {
	state = {
		dailyStepsToggle: true,
		studyHoursToggle: false,
		moviesToggle: false,
	}

	_onDailyStepsPress(){
		const newState = !this.state.dailyStepsToggle;
		this.setState({dailyStepsToggle: newState})
	}
	_onHoursStudyingPress(){
		const newState = !this.state.studyHoursToggle;
		this.setState({studyHoursToggle: newState})
	}
	_onMoviesPress(){
		const newState = !this.state.moviesToggle;
		this.setState({moviesToggle: newState})
	}
	setStyle(background){
		return{
			backgroundColor: background,
			width: 300,
			height: 150,
			flex: 5,
			alignItems: 'center',
			justifyContent: 'center',
			margin: 20,
		}
	}
  render() {

		const dailyStepsText = this.state.dailyStepsToggle
			? "Tracking daily steps: ON"
			: "Tracking daily steps: OFF";
		const hoursStudingText = this.state.studyHoursToggle
			? "Tracking hours of studying: ON"
			: "Tracking hours of studying OFF";
		const moviesText = this.state.moviesToggle
			? "Tracking daily steps: ON"
			: "Tracking daily steps: OFF";
		const dailyStepsBackground = this.state.dailyStepsToggle
			? '#6be069'
			: '#dd6378';
		const hoursStudingBackground = this.state.studyHoursToggle
			? '#6be069'
			: '#dd6378';
		const moviesBackground = this.state.moviesToggle
			? '#6be069'
			: '#dd6378';
    return(
			<View style={styles.container}>
				<TouchableOpacity
					title="Number of steps"
					onPress={() => this._onDailyStepsPress()}
					style={this.setStyle(dailyStepsBackground)}
					>
						<Text style={styles.buttonText}>{dailyStepsText}</Text>
				</TouchableOpacity>
				<TouchableOpacity
					title="Number of steps"
					onPress={() => this._onHoursStudyingPress()}
					style={this.setStyle(hoursStudingBackground)}
					>
						<Text style={styles.buttonText}>{hoursStudingText}</Text>
				</TouchableOpacity>
				<TouchableOpacity
					title="Number of steps"
					onPress={() => this._onMoviesPress()}
					style={this.setStyle(moviesBackground)}
					>
						<Text style={styles.buttonText}>{moviesText}</Text>
				</TouchableOpacity>
			</View>
    );
  }
}
const styles = StyleSheet.create({
	container: {
		flex: 1,
		flexDirection: 'column',
		justifyContent: 'center',
		alignItems: 'center',
	},
	buttonText: {
		fontSize: 16,
	},
})
