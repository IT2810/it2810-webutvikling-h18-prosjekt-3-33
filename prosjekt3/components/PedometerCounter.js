import React from 'react';
import Expo from 'expo';
import { Pedometer } from 'expo';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import * as Storage from './Storage';
import Colors from '../constants/Colors';

//Pedometer component used by StepGoal to get step data from core motion (iOS) or
//Google Fit (Android)
export default class PedometerCounter extends React.Component {
  state = {
    isPedometerActive: "checking",
    pastStepCount: 0,
    currentStepCount: 0,
    stepGoal: 0,
  };
  //Subscribe to pedometer count and setting state from props (from StepGoal)
  componentDidMount() {
    this._subscribe();
    this.setState({
      stepGoal: this.props.stepGoal
    })
  }
  //Unsubscribe from pedometer count on unmount
  componentWillUnmount() {
    this._unsubscribe();
  }
  //State changes on new props from StepGoal
  componentDidUpdate(prevProps, prevState){
    if(this.props.stepGoal != prevProps.stepGoal){
      this.setState({stepGoal: this.props.stepGoal})
    }
  }
//Subscribing to Core Motion and Google FIt
  _subscribe = () => {
    this._subscription = Pedometer.watchStepCount(result => {
      this.setState({
        currentStepCount: result.steps
      });
    });

    Pedometer.isAvailableAsync().then(
      result => {
        this.setState({
          isPedometerActive: String(result)
        });
      },
      error => {
        this.setState({
          isPedometerActive: "Could not get isPedometerActive: " + error
        });
      }
    );

    const end = new Date();
    const start = new Date();
    start.setDate(end.getDate() - 1);
    Pedometer.getStepCountAsync(start, end).then(
      result => {
        this.setState({ pastStepCount: result.steps });
      },
      error => {
        this.setState({
          pastStepCount: "Could not get stepCount: " + error
        });
      }
    );
  };

  _unsubscribe = () => {
    this._subscription && this._subscription.remove();
    this._subscription = null;
  };

  render() {
    //Renders a progress bar based on progress / goal relationship
    var totalProgress = this.state.pastStepCount + this.state.currentStepCount;
    var goal = this.state.stepGoal;
    return (
      <View style={styles.container}>
        <View style={styles.textContainer}>
          <Text>
            {totalProgress} out of {goal}
          </Text>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyle(totalProgress, goal)}>
          </View>
        </View>

      </View>
    );
  }
}
//Function for calculating the progressbar colors and width
progressBarStyle = function(progress, goal){
  let progressWidth = 0;
  let backgroundColor = '#000';
  let progressGoalRelation = progress/goal;
  if(progressGoalRelation>=1){
    progressWidth = 1;
    backgroundColor = '#75CDA8';
  }
  if (progressGoalRelation < 1 && progressGoalRelation > 0.4){
    progressWidth = progressGoalRelation;
    backgroundColor = '#EDBF2C';
  }
  if(progressGoalRelation<=0.4){
    progressWidth = progressGoalRelation;
    backgroundColor = '#DB0200';
  }
  return{
    flex: progressWidth,
    backgroundColor: backgroundColor,
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    marginTop: 15,
    alignItems: "center",
    justifyContent: "center"
  },
  progressBar: {
    flex: 1,
    flexDirection: 'row',
    width: 300,
    height: 20,
    backgroundColor: '#eee',
  },
  textContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    width: 300,
  }
});

Expo.registerRootComponent(PedometerCounter);
