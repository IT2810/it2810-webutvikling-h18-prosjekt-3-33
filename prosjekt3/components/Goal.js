import React from 'react';
import {
  View, Text, StyleSheet, ScrollView

} from 'react-native';
import Colors from '../constants/Colors.js';
import PedometerCounter from './PedometerCounter';

export default class Goal extends React.Component {
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of steps</Text>
          <PedometerCounter />
        </View>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Hours of studing</Text>
        </View>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of movies watched</Text>
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.unfinishedGoalColor,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
