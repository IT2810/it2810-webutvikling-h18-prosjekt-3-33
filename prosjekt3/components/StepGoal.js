import React from 'react';
import {
  View, Text, StyleSheet, ScrollView, Button, TouchableOpacity

} from 'react-native';
import Colors from '../constants/Colors.js';
import PedometerCounter from './PedometerCounter';

export default class StepGoal extends React.Component {
  state = {
    stepGoal: 0,
  }
  componentDidMount(){
    this.setState({stepGoal: this.props.stepGoal})
  }
  componentDidUpdate(prevProps, prevState){
    if(this.props.stepGoal != prevProps.stepGoal){
      this.setState({stepGoal: this.props.stepGoal})
    }
  }
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of steps {this.props.stepGoal} </Text>
          <PedometerCounter stepGoal={this.state.stepGoal} />
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.unfinishedGoalColor,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
