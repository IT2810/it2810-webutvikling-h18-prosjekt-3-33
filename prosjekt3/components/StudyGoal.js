import React from 'react';
import {
  View, Text, StyleSheet, ScrollView, Button, TouchableOpacity

} from 'react-native';
import Colors from '../constants/Colors.js';
import PedometerCounter from './PedometerCounter';
import StudyCounter from './StudyCounter';

export default class StepGoal extends React.Component {
  state = {
    studyGoal: 0,
  }
  componentDidMount(){
    this.setState({studyGoal: this.props.studyGoal})
  }
  componentDidUpdate(prevProps, prevState){
    if(this.props.studyGoal != prevProps.studyGoal){
      this.setState({studyGoal: this.props.studyGoal})
    }
  }
  render() {
    return(
      <ScrollView style={styles.container}>
        <View style={styles.goalContainer}>
          <Text style={styles.goalTitle}>Number of hours studied today </Text>
          <StudyCounter studyGoal={this.props.studyGoal} />
        </View>
      </ScrollView>
    );
  }
}
const styles = StyleSheet.create({
  goalTitle: {
    fontSize: 16,
    color: 'white',
  },
  goalContainer: {
    backgroundColor: Colors.goalComponentOne,
    height: 150,
    padding: 10,
    marginTop: 20
  },
  container: {
    padding: 20
  },
  emptyContainer: {
    height: 80
  }

});
