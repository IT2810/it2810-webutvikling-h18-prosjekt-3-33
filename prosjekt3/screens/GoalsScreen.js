import React from 'react';
import {
	View,
	ScrollView,
	Text,
	StyleSheet,
	TouchableHighlight,
	PropTypes,
	Button,
	TouchableOpacity
} from 'react-native';
import StepGoal from '../components/StepGoal';
import StudyGoal from '../components/StudyGoal';
import PushupsGoal from '../components/PushupsGoal';
import * as Storage from '../components/Storage';

//This is the main component for the goals page
export default class GoalsScreen extends React.Component {
// Setting initial state
	state = {
		stepGoal: 10,
		studyGoal: 10,
		pushupsGoal: 10,
		goalChooser: {
			stepGoal: true,
			studyGoal: false,
			pushupsGoal: false
		}
	}
	// Title for navigation stack
	static navigationOptions = {
		title: 'Daily Progresss',
  };

	//Check if localstorage already has data, else create default values
	componentDidMount(){
		Storage.getGoals().then(goals =>{
			if(goals.hasOwnProperty('stepGoal')){
				this.setState({ stepGoal: goals.stepGoal})
			}else{
				goals.stepGoal = 10000;
				this.setState({stepGoal: 10000})
			}
			if(goals.hasOwnProperty('studyGoal')){
				this.setState({ studyGoal: goals.studyGoal})
			}else{
				goals.studyGoal = 5;
				this.setState({studyGoal: 5})
			}
			if(goals.hasOwnProperty('pushupsGoal')){
				this.setState({ pushupsGoal: goals.pushupsGoal})
			}else{
				goals.pushupsGoal = 5;
				this.setState({pushupsGoal: 5})
			}
			if(goals.hasOwnProperty('goalChooser')){
				this.setState({goalChooser: goals.goalChooser})
			}else{
				goals.goalChooser = {stepGoal: true, studyGoal: true, pushupsGoal: true}
				this.setState({goalChooser: goals.goalChooser})
			}
			Storage.storeGoals(goals);
		});
	}

	//store changes to asyncstorage
	componentDidUpdate(prevProps, prevState){
		if(prevState != this.state){
			Storage.getGoals().then(goals =>{
				goals.stepGoal = this.state.stepGoal;
				goals.studyGoal = this.state.studyGoal;
				goals.pushupsGoal = this.state.pushupsGoal;
				goals.goalChooser = this.state.goalChooser;
				Storage.storeGoals(goals);
			});
		}
	}

	//Callback functions for navigation params
	onLoad = data => {
		this.setState(data);
	}
	onChooseLoad = data => {
		this.setState(data);
	}

  render() {
		//Creating views as const to enable conditional rendering based on goalChooser
		//The navigate is passed data and callback function
		const stepComponent = (
			<TouchableOpacity style={styles.goalView} onPress={() =>this.props.navigation.navigate('EditStep', {stepGoal: this.state.stepGoal, onLoad: this.onLoad})}>
				<StepGoal stepGoal={this.state.stepGoal}/>
			</TouchableOpacity>
		);
		const studyComponent = (
			<TouchableOpacity style={styles.goalView} onPress={() =>this.props.navigation.navigate('EditStudy', {studyGoal: this.state.studyGoal, onLoad: this.onLoad})}>
				<StudyGoal studyGoal={this.state.studyGoal}/>
			</TouchableOpacity>
		);
		const pushupsComponent = (
			<TouchableOpacity style={styles.goalView} onPress={() =>this.props.navigation.navigate('EditPushups', {pushupsGoal: this.state.pushupsGoal, onLoad: this.onLoad})}>
				<PushupsGoal pushupsGoal={this.state.pushupsGoal} />
			</TouchableOpacity>
		);
    return (
			<ScrollView style={styles.container}>
				<Button
					title="Choose goals"
					onPress={() => this.props.navigation.navigate('EditGoals',
						{goalChooser: this.state.goalChooser, onLoad: this.onChooseLoad})}/>
				{this.state.goalChooser.stepGoal ? stepComponent : null}
				{this.state.goalChooser.studyGoal ? studyComponent : null}
				{this.state.goalChooser.pushupsGoal ? pushupsComponent : null}
			</ScrollView>
		);
  }

}


const styles = StyleSheet.create({
	container: {
		flex: 1,
		flexDirection: 'column',
	},
	goalView: {
		marginBottom: -40,
	},
});
