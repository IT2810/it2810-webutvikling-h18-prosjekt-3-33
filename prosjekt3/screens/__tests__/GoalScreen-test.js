import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import GoalsScreen from '../GoalsScreen';
import * as Storage from '../../components/Storage';
import StepGoal from '../../components/StepGoal.js';
import StudyGoal from '../../components/StudyGoal.js';
import PushupsGoal from '../../components/PushupsGoal.js';
import EditGoalScreen from '../SubGoalScreens/EditGoalScreen.js';
import ShallowRenderer from 'react-test-renderer/shallow';
import {shallow} from 'enzyme';
import ToggleSwitch from '../SubGoalScreens/ToggleSwitch.js';

describe("GoalsScreen Testing", () => {

  it('renders the GoalScreen', async () => {
      const tree = renderer.create(<GoalsScreen />).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('renders the StepGoal component', async () => {
      const tree = renderer.create(<StepGoal />).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('renders the StudyGoal component', async () => {
      const tree = renderer.create(<StudyGoal />).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('renders the PushupsGoal component', async () => {
      const tree = renderer.create(<PushupsGoal />).toJSON();
      expect(tree).toMatchSnapshot();
  });

  it('calling the onChooseLoad-function in GoalScreen', () => {
      let GoalsScreenComponent = renderer.create(<GoalsScreen />).getInstance();
      GoalsScreenComponent.onChooseLoad({goalChooser: {
  			stepGoal: true,
  			studyGoal: false,
  			pushupsGoal: false
  		}});

      let item = {
  			stepGoal: true,
  			studyGoal: false,
  			pushupsGoal: false
  		}
      expect(GoalsScreenComponent.state.goalChooser).toEqual(item);

  });

  it('calling the onLoad-function in GoalScreen', () => {
      let GoalsScreenComponent = renderer.create(<GoalsScreen />).getInstance();
      GoalsScreenComponent.onLoad({stepGoal: 7000, studyGoal: 6, pushupsGoal: 10});

      let stepGoalItem = 7000;
      let studyGoalItem = 6;
      let pushupsGoalItem = 10;
      expect(GoalsScreenComponent.state.stepGoal).toEqual(stepGoalItem);
      expect(GoalsScreenComponent.state.studyGoal).toEqual(studyGoalItem);
      expect(GoalsScreenComponent.state.pushupsGoal).toEqual(pushupsGoalItem);

  });
});
