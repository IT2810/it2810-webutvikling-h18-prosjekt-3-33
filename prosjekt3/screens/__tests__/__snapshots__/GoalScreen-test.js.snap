// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsScreen Testing Snapshot test for EditGoalScreen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 50,
      }
    }
  >
    <ToggleSwitch
      isOn={undefined}
      label="Track number of steps"
      labelStyle={
        Object {
          "color": "black",
          "fontWeight": "900",
        }
      }
      offColor="red"
      onColor="green"
      onToggle={[Function]}
      size="large"
    />
    ;
  </View>
  <View
    style={
      Object {
        "marginBottom": 50,
      }
    }
  >
    <ToggleSwitch
      isOn={undefined}
      label="Track hours of studying"
      labelStyle={
        Object {
          "color": "black",
          "fontWeight": "900",
        }
      }
      offColor="red"
      onColor="green"
      onToggle={[Function]}
      size="large"
    />
    ;
  </View>
  <View>
    <ToggleSwitch
      isOn={undefined}
      label="Track number of pushups"
      labelStyle={
        Object {
          "color": "black",
          "fontWeight": "900",
        }
      }
      offColor="red"
      onColor="green"
      onToggle={[Function]}
      size="large"
    />
    ;
  </View>
</View>
`;

exports[`GoalsScreen Testing Snapshot test for EditStepGoal 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e8fcb8",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "width": 300,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "padding": 10,
        }
      }
    >
      Set a new goal for daily number of steps
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 32,
        }
      }
    />
  </View>
</View>
`;

exports[`GoalsScreen Testing Snapshot test for EditStudyGoal 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e8fcb8",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "width": "80%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Set a new goal for daily hours of studying 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 32,
        }
      }
    />
  </View>
</View>
`;

exports[`GoalsScreen Testing renders the GoalScreen 1`] = `null`;

exports[`GoalsScreen Testing renders the PushupsGoal component 1`] = `null`;

exports[`GoalsScreen Testing renders the StepGoal component 1`] = `null`;

exports[`GoalsScreen Testing renders the StudyGoal component 1`] = `null`;
